lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -125,35 +151,89 @@ withNetworkLayer tr project k = k NetworkLayer
-    parseBlockHeader blockHash =
-        case fromText (BF.unBlockHash blockHash) of
-            Right hash -> pure hash
-            Left tde -> throwError $ InvalidBlockHash blockHash tde
+        fromBlockfrost
+            <=< ExceptT
+            <<< (first ClientError <$>)
+            <<< BF.runBlockfrostClientT project
+
+class FromBlockfrost b w where
+    fromBlockfrost :: forall m. MonadError BlockfrostError m => b -> m w
+
+instance FromBlockfrost BF.Block BlockHeader where
+    fromBlockfrost block@BF.Block{..} = do
+        slotNo <- case _blockSlot of
+            Just s -> pure $ SlotNo $ fromIntegral $ BF.unSlot s
+            Nothing -> throwError $ NoSlotError block
+        blockHeight <- case _blockHeight of
+            Just height -> pure $ Quantity $ fromIntegral height
+            Nothing -> throwError $ NoBlockHeight block
+        headerHash <- parseBlockHeader _blockHash
+        parentHeaderHash <- for _blockPreviousBlock parseBlockHeader
+        pure BlockHeader { slotNo, blockHeight, headerHash, parentHeaderHash }
+      where
+        parseBlockHeader blockHash =
+            case fromText (BF.unBlockHash blockHash) of
+                Right hash -> pure hash
+                Left tde -> throwError $ InvalidBlockHash blockHash tde
+
+instance FromBlockfrost BF.ProtocolParams ProtocolParameters where
+    fromBlockfrost BF.ProtocolParams{..} = do
+        decentralizationLevel <-
+            let percentage = mkPercentage $
+                    toRational _protocolParamsDecentralisationParam
+            in case percentage of
+                Left PercentageOutOfBoundsError ->
+                    throwError $ InvalidDecentralizationLevelPercentage
+                        _protocolParamsDecentralisationParam
+                Right level -> pure $ DecentralizationLevel level
+        let intToQuantity :: (Num q, Integral i) => i -> Quantity s q
+            intToQuantity = Quantity . fromIntegral
+        let txParameters = TxParameters
+                { getFeePolicy =  LinearFee
+                    (intToQuantity _protocolParamsMinFeeA)
+                    (intToQuantity _protocolParamsMinFeeB)
+                , getTxMaxSize =
+                    intToQuantity _protocolParamsMaxTxSize
+                , getTokenBundleMaxSize =
+                    TokenBundleMaxSize (TxSize 0) -- TODO: where to get it from?
+                , getMaxExecutionUnits =
+                    ExecutionUnits
+                        { executionSteps = fromIntegral $
+                            BF.unQuantity _protocolParamsMaxTxExSteps
+                        , executionMemory = fromIntegral $
+                            BF.unQuantity _protocolParamsMaxTxExMem
+                        }
+                }
+            desiredNumberOfStakePools = fromIntegral _protocolParamsNOpt
+        let minimumUTxOvalue =
+                MinimumUTxOValue $ Coin (fromIntegral _protocolParamsMinUtxo)
```

Comment:
You probably want to use https://github.com/input-output-hk/cardano-wallet/blob/ace50587ad9f0dd1f066117c7e653ed9486d97da/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1092-L1095
and Blockfrost's `coins_per_utxo_word` instead.